src/System.Slices/tools/ILSub/ILSub/Program.cs

```diff
@@ -275,16 +273,19 @@ static string RewriteIL(string input)
             while (i < input.Length)
             {
                 Match methodBody;
-                Match constructorBytes;
-                if (!FindMethodWithILSubInstance(input, i, input.Length, out methodBody, out constructorBytes))
+                Match ilText;
+                if (!FindMethodWithILSubInstance(input, i, input.Length, out methodBody, out ilText))
                 {
                     sb.Append(input, i, input.Length - i);
                     return sb.ToString();
                 }
 
                 sb.Append(input, i, methodBody.Index - i);
-                byte[] bytes = GetBytes(input, constructorBytes);
-                sb.Append(Encoding.UTF8.GetString(bytes, 4, bytes.Length - 6));
+
+                var ilSb = new StringBuilder(input, ilText.Index, ilText.EndIndex - ilText.Index, ilText.EndIndex - ilText.Index);
```

Comment:
If GetBytes/IsHex/HexToByte is no longer used I'd remove it. I didn't know about the caverbal